File: backend/src/main/java/com/ifpb/agendapet/appointment/dto/AppointmentResponseDTO.java
package com.ifpb.agendapet.appointment.dto;

import com.ifpb.agendapet.appointment.AppointmentStatus;
import com.ifpb.agendapet.appointment.AppointmentTypes;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record AppointmentResponseDTO(UUID id, UUID doctor_id, String doctor_name, UUID pet_id, String pet_name, LocalDateTime scheduled_at, AppointmentTypes type, AppointmentStatus status, BigDecimal price, com.ifpb.agendapet.shared.enums.PaymentStatus paymentStatus, String note, LocalDateTime created_at) {
}

File: frontend/src/app/models/DTO/appointment-response-DTO.ts
export interface AppointmentResponseDTO {
  id: string;
  doctor_id: string;
  doctor_name: string;
  pet_id: string;
  pet_name: string;
  scheduled_at: string;
  type: AppointmentTypes;
  status: AppointmentStatus;
  price: number;
  paymentStatus: PaymentStatus;
  note?: string;
  created_at: string;
}

export enum AppointmentTypes {
  CONSULTATION = 'CONSULTATION',
  VACCINATION = 'VACCINATION',
  SURGERY = 'SURGERY',
}

export enum AppointmentStatus {
  SCHEDULED = 'SCHEDULED',
  COMPLETED = 'COMPLETED',
  CANCELED = 'CANCELED',
  NO_SHOW = 'NO_SHOW',
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  PAID = 'PAID',
  REFUNDED = 'REFUNDED',
}
